dali/pipeline/executor/executor.h

```diff
@@ -478,25 +484,26 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunMixed() {
   // iterations of a stage of the pipeline.
   CUDA_CALL(cudaEventSynchronize(mixed_stage_event_));
 
-  try {
     for (int i = 0; i < graph_->NumOp(OpType::MIXED); ++i) {
       OpNode &op_node = graph_->Node(OpType::MIXED, i);
-      typename WorkspacePolicy::template ws_t<OpType::MIXED> ws =
-          WorkspacePolicy::template GetWorkspace<OpType::MIXED>(mixed_idxs, *graph_, i);
-      TimeRange tr("[Executor] Run Mixed op " + op_node.instance_name,
-          TimeRange::kOrange);
-      RunHelper(op_node, ws);
-      FillStats(mixed_memory_stats_, ws,  "MIXED_" + op_node.instance_name,
-                mixed_memory_stats_mutex_);
-      if (ws.has_stream() && ws.has_event()) {
-        CUDA_CALL(cudaEventRecord(ws.event(), ws.stream()));
+      try {
+        typename WorkspacePolicy::template ws_t<OpType::MIXED> ws =
+            WorkspacePolicy::template GetWorkspace<OpType::MIXED>(mixed_idxs, *graph_, i);
+        TimeRange tr("[Executor] Run Mixed op " + op_node.instance_name,
+            TimeRange::kOrange);
+        RunHelper(op_node, ws);
+        FillStats(mixed_memory_stats_, ws,  "MIXED_" + op_node.instance_name,
+                  mixed_memory_stats_mutex_);
+        if (ws.has_stream() && ws.has_event()) {
+          CUDA_CALL(cudaEventRecord(ws.event(), ws.stream()));
+        }
+        CUDA_CALL(cudaGetLastError());
```

Comment:
I am just curious, wouldn't `CUDA_CALL(cudaEventRecord(ws.event(), ws.stream()));` trigger the error as well?

Reply:
```
- Note that this function may also return error codes from previous, asynchronous launches.
- Note that this function may also return cudaErrorInitializationError, cudaErrorInsufficientDriver or cudaErrorNoDevice if this call tries to initialize internal CUDA RT state.
```
It could

Reply:
It would, but it's in `if` statement, I guess workspace not having a stream would be odd, but I want to have not-conditional check here.